lib/fs.js

```diff
@@ -2034,6 +2034,11 @@ ReadStream.prototype.close = function(cb) {
   this.fd = null;
 };
 
+// needed because as it will be called with arguments
+// that does not match this.close() signature
+function closeOnOpen() {
+  this.close();
```

Comment:
`EventEmitter` instances always call their event handlers within the context of the `EventEmitter`. I'm not sure we need to explicitly document that in this one particular place when we do this same thing throughout core.